src/middlewared/middlewared/plugins/pool.py

```diff
@@ -4182,6 +4182,147 @@ def __handle_zfs_set_property_error(self, e, properties_definitions):
             raise CallError(pjob.error)
         return data
 
+    @accepts(
+        Str('id', required=True),
+        Str('type', enum=['USER', 'GROUP']),
+        Ref('query-filters'),
+        Ref('query-options'),
+    )
+    @item_method
+    async def get_userquota(self, id, type, filters, options):
+        """
+        Return a list of the specified `type` of  quotas on the ZFS dataset `id`.
+        Support `query-filters` and `query-options`. used_bytes and used_percentage
+        may not instantly update as space is used.
+
+        Each quota entry has these fields:
+
+        `id` - the uid or gid to which the quota applies.
+
+        `name` - the user or group name to which the quota applies. Value is
+        null if the id in the quota cannot be resolved to a user or group. This
+        indicates that the user or group does not exist on the server.
+
+        `quota` - the quota size in bytes.
+
+        `used_bytes` - the amount of bytes the user has written to the dataset.
+
+        `used_percentage` - the percentage of the user or group quota consumed.
+        """
+        dataset = (await self._get_instance(id))['name']
+        quota_list = []
+        quota_get = await run(
+            ['zfs', f'{type.lower()}space', '-H', '-n', '-p', '-o', 'name,used,quota', dataset],
```

Comment:
We should be making `zfs` specific calls in zfs plugin please. I forgot about it earlier.